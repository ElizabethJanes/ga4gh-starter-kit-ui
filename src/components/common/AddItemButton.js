import React from 'react';
import { 
    Container,
    Typography, 
    FormControl,
    TextField, 
    Grid, 
    FormGroup, 
    Button, 
    Divider, 
    Box, 
    RadioGroup, 
    Radio, 
    FormControlLabel, 
    IconButton, 
    MenuItem, 
    Tooltip,
    InputAdornment
} from '@material-ui/core';
import AddCircle from '@material-ui/icons/AddCircle';

/*
    Displays a "+" icon and can be used to add additional fields for list-type
    items. It is passed a handler function as a prop, which should append
    another instance of the item to the list of items being updated.
*/
const AddItemButton = props => {
    return (
        <div>
            {props.display
                ?
                    <Grid container>
                        <Grid item>
                            <Tooltip title={`Add ${props.objectName}`}>
                                <IconButton color='primary' onClick={props.handleClick} disabled={props.disabled}>
                                    <AddCircle/>
                                </IconButton>   
                            </Tooltip>    
                        </Grid>
                    </Grid>
                : null
            }
        </div>
    )
}

export default AddItemButton;