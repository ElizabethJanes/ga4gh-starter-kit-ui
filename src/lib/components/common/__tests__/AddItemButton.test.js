import React from 'react';
import renderer from 'react-test-renderer';
import AddItemButton from '../AddItemButton';

test('SHOW <AddItemButton />', () => {
    const addItemButton = renderer
    .create(<AddItemButton readOnly={true} objectName={'SHOW snapshot test'} />)
    .toJSON();
    expect(addItemButton).toMatchSnapshot();
});

test('NEW and EDIT <AddItemButton />', () => {
    const addItemButton = renderer
    .create(<AddItemButton readOnly={false} objectName={'NEW snapshot test'} />)
    .toJSON();
    expect(addItemButton).toMatchSnapshot();
});