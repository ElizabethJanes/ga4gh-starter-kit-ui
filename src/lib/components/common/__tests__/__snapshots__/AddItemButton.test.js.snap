// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <AddItemButton /> 1`] = `<div />`;

exports[`SHOW <AddItemButton /> 1`] = `<div />`;
